shared/api/minja.hpp

```diff
@@ -0,0 +1,3940 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// The following code is based on the minja project (https://github.com/google/minja).
+// with modifications to be more VS and mobile compiler friendly.
+/*
+    Copyright 2024 Google LLC
+
+    Use of this source code is governed by an MIT-style
+    license that can be found in the LICENSE file or at
+    https://opensource.org/licenses/MIT.
+*/
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <regex>
+#include <memory>
+#include <stdexcept>
+#include <sstream>
+#include <unordered_set>
+
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::ordered_json;
+
+namespace minja
```

Comment:
having minja impl locally rather than adding a dependency makes for easier msft privacy/security compliance so that's good - how do we plan on maintaining this though? will need intermittent checks monitoring minja PRs to ensure parity with official minja in case of changes, or something similar.

Reply:
the current code already made some changes from its original version. In the future, we need to maintain this file by ourselves

Reply:
sounds good!